hbase-server/src/main/java/org/apache/hadoop/hbase/wal/AbstractRecoveredEditsOutputSink.java

```diff
@@ -73,33 +73,27 @@ protected RecoveredEditsWriter createRecoveredEditsWriter(TableName tableName, b
     return new RecoveredEditsWriter(region, regionEditsPath, w, seqId);
   }
 
-  protected Path closeRecoveredEditsWriter(RecoveredEditsWriter editsWriter,
+  /**
+   * abortRecoveredEditsWriter closes the editsWriter, but does not rename and finalize the
+   * recovered edits WAL files. Please see HBASE-28569.
+   */
+  protected void abortRecoveredEditsWriter(RecoveredEditsWriter editsWriter,
     List<IOException> thrown) throws IOException {
-    try {
-      editsWriter.writer.close();
-    } catch (IOException ioe) {
-      final String errorMsg = "Could not close recovered edits at " + editsWriter.path;
-      LOG.error(errorMsg, ioe);
-      updateStatusWithMsg(errorMsg);
-      thrown.add(ioe);
+    closeRecoveredEditsWriter(editsWriter, thrown);
+    if (editsWriter.editsWritten == 0) {
+      // just remove the empty recovered.edits file
+      removeRecoveredEditsFile(editsWriter);
+    }
```

Comment:
The recovered edits file is incomplete at this point, correct? Should we just always delete it?

Reply:
I think we can always delete it.